src/processor.ts

```diff
@@ -0,0 +1,35 @@
+import {FileHandler} from './file-handler';
+import {Transformer} from './transformer';
+import {Validator} from './validator';
+
+/**
+ * This class is responsible for injecting dependencies and connecting the relevant
+ * classes together, delegating requests to the correct place.
+ */
+
+export class Processor {
+  validate(filename: string) {
```

Comment:
Looks like `filename` is common to all methods here. Maybe use a constructor instead of passing them to the methods!

Reply:
i've re-written this class to make the intention behind it clearer - essentially it's kind of like a singleton which is responsible for instantiating and then injecting classes into the various `Operation`s which i've now laid out below. I've also added a stub `build` method to demonstrate that there are different methods that don't always take the same parameters. So to me passing in a `filename` to the constructor doesn't feel quite right, but hopefully the way it's written now that makes more sense